### ISSUE TITLE
Version 7.1.3 breaks magento2 deploy:vendors

### ISSUE THREAD

**Issue by wilfriedwolf** [deployphp/deployer#3454]
- Deployer version:  7.1.3
- Deployment OS: debian/docker

Trying to build an artifact from a magento2 installation, the deploy:vendors command fails. As I see it (after debugging for a while) in `[localhost] > post-install-cmd: MagentoHackathon\Composer\Magento\Plugin->onNewCodeEvent` an exception is thrown. Then dep fails with

```
include(phar:///var/www/vhosts/.../vendor/deployer/deployer/dep.phar/vendor/composer/../../src/Exception/GracefulShutdownException.php): Failed to open stream: phar error: Cannot open phar archive "/var/www/vhosts/.../vendor/deployer/deployer/dep.phar" for reading
```
and the Message  `ErrorException  in ClassLoader.php on line 578:`

Exactly the same deployment works with v7.1.1

**Comment by antonmedv** [comment]
>  Cannot open phar archive "/var/www/vhosts/.../vendor/deployer/deployer/dep.phar" for reading

This is strange.

**Comment by wilfriedwolf** [comment]
It is! I think I could break it down further, for now I fixed my deployment to version 7.1.1...

**Comment by wilfriedwolf** [comment]
Hm, I could debug inside the PHAR yesterday but it passes all breakpoints today :thinking:

**Comment by akosglue** [comment]
I encountered it also on 7.1.1

**Comment by akosglue** [comment]
@wilfriedwolf how did you fix it to version 7.1.1.?

**Comment by wilfriedwolf** [comment]
@akosglue I did not fix it in version 7.1.1, the error just did not apear in this version. Interesting enough just a week ago, there was no error with version 7.1.4 deployment worked correct. Then today, same version, old error.
@antonmedv there should not be different packages for the same tag, shouldn't they?

**Comment by antonmedv** [comment]
dev-master and tages (like v7.1.4) are different. Tags are "bundled". See https://github.com/deployphp/deployer/tree/v7.1.4

**Comment by wilfriedwolf** [comment]
I tried v7.2.0 before debugging, and this one works, hopefully for good.

**Comment by wilfriedwolf** [comment]
Well, idiot me..

To be able to  deploy I installed it with `COMPOSER=deploy.json composer install ...` . since when there is no `deploy.lock` when releasing a new deployer version, it became the new one, whereas in the `composer.lock` there was still an older version. So calling `dep deploy:vendor` replaced the `dep(.phar)` which results in the above seen error (not being able to open phar for reading).

I solved this by just using a globally installed deployer in the docker containers for deploying, now I am hopefully good with it.
